### PR TITLE
hasura: 1.0.0 -> 1.1.1

### PR DESCRIPTION
**Pull request by disassembler**
* refactors graphql-engine

### CODE REVIEW

File: nix/nixos/cardano-graphql-service.nix [hunk 1]
```diff
@@ -62,14 +62,14 @@ in {
...
-    hasuraBaseUri = cfg.hasuraProtocol + "://" + cfg.hasuraIp + ":" + (toString cfg.enginePort) + "/";
+    hasuraBaseUri = cfg.hasuraProtocol + "://" + cfg.hasuraIp + ":" + (toString cfg.enginePort);
     hasuraDbMetadata = ../../hasura/migrations/metadata.json;
   in lib.mkIf cfg.enable {
     systemd.services.cardano-graphql = {
       wantedBy = [ "multi-user.target" ];
       requires = [ "graphql-engine.service" ];
       environment = {
-        HASURA_URI = hasuraBaseUri + "v1/graphql";
+        HASURA_URI = hasuraBaseUri;
```
Comment (rhyslbw) [inline nix/nixos/cardano-graphql-service.nix:72]:
```suggestion
        HASURA_URI = hasuraBaseUri + "v1/graphql";
```

File: nix/nixos/graphql-engine-service.nix [hunk 2]
```diff
@@ -82,6 +83,7 @@ in {
           -d ${cfg.db} \
           --port ${toString cfg.dbPort} \
           serve \
+          ${lib.optionalString cfg.filterHasuraOperations "--enable-allowlist"} \
```
Comment (rhyslbw) [inline nix/nixos/graphql-engine-service.nix:86]: We will no longer be using the allow-list feature of Hasura

File: nix/packages.nix [hunk 3]
```diff
@@ -7,7 +7,28 @@
...
+        in_blacklist =
+          lib.hasPrefix "/node_modules" sansPrefix ||
+          lib.hasPrefix "/build" sansPrefix;
+        in_whitelist =
+          (type == "directory") ||
+          (lib.hasSuffix ".yml" name) ||
+          (lib.hasSuffix ".ts" name) ||
+          (lib.hasSuffix ".json" name) ||
+          (lib.hasSuffix ".graphql" name) ||
+          (lib.hasPrefix "/source" sansPrefix) ||
```
Comment (rhyslbw) [inline nix/packages.nix:23]:
```suggestion
          (lib.hasPrefix "/src" sansPrefix) ||
```

Comment (disassembler) [reply nix/packages.nix:23]: I'll just remove it, obviously not needed since it builds without all of src being whitelisted.

**Pull request merged**